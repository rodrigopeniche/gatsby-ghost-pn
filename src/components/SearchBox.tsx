import * as React from "react"

const SearchBox = (props: SearchProps) => (
    <form data-testid="numbers-form">
        {"+1 ("}
        <input
            // type="number" //cant be number with alpphanumeric characters
            name="areaCode"
            size={4}
            defaultValue={props.areaCode}
        />
        {")"}
        <input
            // type="number"
            name="exchange"
            size={4}
            defaultValue={props.exchange}
        />
        <input
            // type="number"
            name="subscriber1"
            size={2}
            defaultValue={props.subscriber1}
        />
        {"--"}
        <input
            // type="number"
            name="subscriber2"
            size={2}
            defaultValue={props.subscriber2}
        />
    </form>
)

interface SearchProps {
    areaCode?: string | number
    exchange?: string | number
    subscriber1?: string | number
    subscriber2?: string | number
}

SearchBox.defaultProps = {
    areaCode: "XXX",
    exchange: "XXX",
    subscriber1: "XX",
    subscriber2: "XX",
}

export default SearchBox
