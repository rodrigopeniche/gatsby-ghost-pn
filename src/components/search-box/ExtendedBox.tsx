import * as React from "react"
import ExtendedContainer from "./extended/ExtendedContainer"
import Label from "./extended/Label"
import KnownScams from "./extended/KnownScams"
import Results from "./extended/Results"
import AreaCode from "./extended/AreaCode"
import PrefixCode from "./extended/PrefixCode"
import Warning from "./extended/Warning"
import ProceedButton from "./extended/ProceedButton"
import Underlined from "./extended/Underlined"

const SearchBoxExtended = ({ expanded }: { expanded: boolean }) => {
    if (!expanded) return null
    return (
        <ExtendedContainer>
            <Underlined>
                <Label />
                <AreaCode />
                <PrefixCode />
                <KnownScams />
            </Underlined>
            <Results />
            <Warning />
            <ProceedButton />
        </ExtendedContainer>
    )
}

export default SearchBoxExtended
