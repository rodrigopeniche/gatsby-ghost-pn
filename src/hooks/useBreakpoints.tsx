import * as React from "react"
import { useState, useEffect, createContext, useContext } from "react"

const defaultValue = {}

const BreakpointContext = createContext(defaultValue)

const BreakpointProvider = ({
    children,
    queries,
}: {
    children: React.ReactNode | React.ReactNode[]
    queries: { [key: string]: string }
}) => {
    const [queryMatch, setQueryMatch] = useState({})

    useEffect(() => {
        const mediaQueryLists: { [key: string]: any } = {}
        const keys = Object.keys(queries)
        let isAttached = false

        const handleQueryListener = () => {
            const updatedMatches = keys.reduce(
                (acc: { [key: string]: any }, media) => {
                    acc[media] = !!(
                        mediaQueryLists[media] && mediaQueryLists[media].matches
                    )
                    return acc
                },
                {}
            )
            setQueryMatch(updatedMatches)
        }

        if (window && window.matchMedia) {
            const matches: { [key: string]: any } = {}
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media] = window.matchMedia(queries[media])
                    matches[media] = mediaQueryLists[media].matches
                } else {
                    matches[media] = false
                }
            })
            setQueryMatch(matches)
            isAttached = true
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media].addListener(handleQueryListener)
                }
            })
        }

        return () => {
            if (isAttached) {
                keys.forEach(media => {
                    if (typeof queries[media] === "string") {
                        mediaQueryLists[media].removeListener(
                            handleQueryListener
                        )
                    }
                })
            }
        }
    }, [queries])

    return (
        <BreakpointContext.Provider value={queryMatch}>
            {children}
        </BreakpointContext.Provider>
    )
}

function useBreakpoint(): { [key: string]: boolean } {
    const context = useContext(BreakpointContext)
    if (context === defaultValue) {
        throw new Error("useBreakpoint must be used within BreakpointProvider")
    }
    return context
}
export { useBreakpoint, BreakpointProvider }
