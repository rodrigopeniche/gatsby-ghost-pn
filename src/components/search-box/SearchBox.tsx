import * as React from "react"
import CollapsedBox from "./CollapsedBox"
import ExtendedBox from "./ExtendedBox"

const Container = ({ children }: { children: React.ReactNode[] }) => {
    const containerStyles = {
        minWidth: "200px",
        gridArea: "search-box",
    }

    return <div style={containerStyles}>{children}</div>
}

const SearchBox = () => {
    const [expanded, setExpand] = React.useState(false)
    return (
        <Container>
            <CollapsedBox setExpand={setExpand} />
            <ExtendedBox expanded={expanded} />
        </Container>
    )
}

export default SearchBox
