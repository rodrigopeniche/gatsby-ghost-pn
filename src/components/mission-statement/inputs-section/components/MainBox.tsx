import * as React from "react"
import "./RemoveArrows.css"
import { useBreakpoint } from "../../../../hooks/useBreakpoints"

const MainBox = ({
    name,
    value,
    inputHandler,
}: {
    name: string
    value: string
    inputHandler(name: string, value: string): void
}) => {
    const { md } = useBreakpoint()

    const mainBoxesStyles: React.CSSProperties = {
        maxWidth: "70px",
        minWidth: "50px",
        boxSizing: "border-box",
        height: "33px",
        marginTop: "0px",
        textAlign: "center",
        background: "#FFFAC1",
        border: "none",
        borderRadius: "5px",
        marginRight: "2px",
        marginLeft: "2px",
        marginBottom: "0px",
        display: "inline-block",
        ...(md && { marginRight: "4px", marginLeft: "4px" }),
    }

    return (
        <input
            type="number"
            name={name}
            size={4}
            value={value?.substr(0, 4) || ""}
            style={mainBoxesStyles}
            onChange={e => inputHandler(name, e.target.value)}
            required
        />
    )
}

export default MainBox
