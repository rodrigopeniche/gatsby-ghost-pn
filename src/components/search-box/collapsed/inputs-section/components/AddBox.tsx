import * as React from "react"
import "./RemoveArrows.css"
import { useBreakpoint } from "../../../../../hooks/useBreakpoints"

const AddBox = ({
    name,
    value,
    inputHandler,
    setHover,
}: {
    name: string
    value: string
    inputHandler(name: string, value: string): void
    setHover: Function
}) => {
    const { md } = useBreakpoint()

    const additionalBoxesStyles: React.CSSProperties = {
        maxWidth: "59px",
        minWidth: "39px",
        boxSizing: "border-box",
        height: "33px",
        marginTop: "0px",
        background: "#FFFAC1",
        border: "none",
        borderRadius: "5px",
        marginRight: "5px",
        marginLeft: "5px",
        marginBottom: "0px",
        display: "inline-block",
        textAlign: "center",
        ...(md && {
            marginRight: "4px",
            marginLeft: "4px",
        }),
    }

    return (
        <input
            type="number"
            name={name}
            size={2}
            value={value?.substr(0, 2) || ""}
            style={additionalBoxesStyles}
            onChange={e => inputHandler(name, e.target.value)}
            onMouseEnter={() => setHover(true)}
            onMouseLeave={() => setHover(false)}
            required
            disabled
        />
    )
}

export default AddBox
