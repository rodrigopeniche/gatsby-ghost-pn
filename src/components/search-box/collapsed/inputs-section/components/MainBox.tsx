import * as React from "react"
import "./RemoveArrows.css"
import { useBreakpoint } from "../../../../../hooks/useBreakpoints"

const MainBox = ({
    name,
    value,
    inputHandler,
    setHover,
}: {
    name: string
    value: string
    inputHandler(name: string, value: string): void
    setHover: Function
}) => {
    const { md } = useBreakpoint()

    const mainBoxesStyles: React.CSSProperties = {
        maxWidth: "70px",
        minWidth: "50px",
        boxSizing: "border-box",
        height: "33px",
        marginTop: "0px",
        background: "#FFFAC1",
        border: "none",
        borderRadius: "5px",
        marginRight: "8px",
        marginLeft: "8px",
        marginBottom: "0px",
        display: "inline-block",
        textAlign: "center",
        ...(md && { marginRight: "4px", marginLeft: "4px" }),
    }

    return (
        <input
            type="number"
            name={name}
            size={3}
            value={value?.substr(0, 3) || ""}
            style={mainBoxesStyles}
            onChange={e => inputHandler(name, e.target.value)}
            onMouseEnter={() => setHover(true)}
            onMouseLeave={() => setHover(false)}
            disabled
            required
        />
    )
}

export default MainBox
