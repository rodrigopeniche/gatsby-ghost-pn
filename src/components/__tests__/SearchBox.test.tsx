import * as React from "react"
import * as renderer from "react-test-renderer"
import "@testing-library/jest-dom/extend-expect"
import SearchBox from "../search-box/SearchBox"
import { BreakpointProvider } from "../../hooks/useBreakpoints"
import AddBox from "../search-box/collapsed/inputs-section/components/AddBox"
import MainBox from "../search-box/collapsed/inputs-section/components/MainBox"
import ExtendedBox from "../search-box/ExtendedBox"

const queries = {
    md: "(max-width: 767px)",
    sm: "(max-width: 600px)",
    xs: "(max-width: 480px)",
    ms: "(max-width: 420px)",
    ss: "(max-width: 320px)",
}

describe("SearchBox", () => {
    const searchBox = (
        <BreakpointProvider queries={queries}>
            <SearchBox />
        </BreakpointProvider>
    )

    const component = renderer.create(searchBox)

    it("renders correctly", () => {
        const jsonTree = component.toJSON()
        expect(jsonTree).toMatchSnapshot()
    })

    it("have number inputs", () => {
        const addBoxes = component.root.findAllByType(AddBox)
        const mainBoxes = component.root.findAllByType(MainBox)
        expect(addBoxes.length).toBe(2)
        expect(mainBoxes.length).toBe(2)
    })

    it("changes input should expand searchbox", async () => {
        const input = component.root.findAllByType("input").pop()
        const extended = component.root.findByType(ExtendedBox)

        renderer.act(() => {
            input.props.onChange({ target: { value: "test" } })
        })

        expect(extended.props.expanded).toBeTruthy()
    })
})
