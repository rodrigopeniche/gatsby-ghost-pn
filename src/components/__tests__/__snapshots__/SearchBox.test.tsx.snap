// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox elements testing renders correctly 1`] = `
<div
  style={
    Object {
      "gridArea": "search-box",
      "minWidth": "200px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#315680",
        "backgroundImage": "url(undefined)",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "boxSizing": "border-box",
        "marginLeft": "11px",
        "marginRight": "18px",
        "maxWidth": "633px",
        "minHeight": "260px",
        "paddingLeft": "16px",
        "paddingRight": "16px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "block",
          "paddingTop": "41px",
        }
      }
    >
      <h1
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Georgia",
            "fontSize": "36px",
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginBottom": "22px",
            "textAlign": "center",
          }
        }
      >
        Select a number
      </h1>
      <div
        style={
          Object {
            "borderBottom": "1.3px solid #FFFFFF",
            "opacity": "0.23",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexWrap": "wrap",
          "fontFamily": "Arial",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "justifyContent": "center",
          "margin": "0 auto",
          "paddingTop": "18px",
          "position": "relative",
          "textAlign": "center",
        }
      }
    >
      <p
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#FFFFFF",
            "display": "inline-block",
            "fontSize": "30px",
            "lineHeight": "34px",
            "marginBottom": "0px",
            "marginRight": "10px",
            "verticalAlign": "middle",
          }
        }
      >
        +1
      </p>
      <div
        style={
          Object {
            "display": "flex",
            "flexBasis": "58px",
            "flexGrow": 1,
            "maxWidth": "104px",
            "minWidth": "58px",
            "position": "relative",
          }
        }
      >
        <p
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#FFFFFF",
              "display": "none",
              "fontSize": "12px",
              "left": "-14px",
              "lineHeight": "34px",
              "marginBottom": "0px",
              "position": "absolute",
              "verticalAlign": "middle",
            }
          }
        >
          +1
        </p>
        <p
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#FFFFFF",
              "fontFamily": "Arial",
              "fontSize": "30px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": "34px",
              "marginBottom": "0px",
            }
          }
        >
          (
        </p>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "maxWidth": "84px",
              "width": "100%",
            }
          }
        >
          <label
            style={
              Object {
                "color": "#FFFFFF",
                "display": "block",
                "fontFamily": "Arial",
                "fontSize": "12px",
                "fontStyle": "normal",
                "marginBottom": "0px",
                "paddingBottom": "8px",
                "textAlign": "center",
              }
            }
          >
            Area code
          </label>
          <input
            name="area-code"
            onChange={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            required={true}
            size={3}
            style={
              Object {
                "background": "#FFFAC1",
                "border": "none",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "33px",
                "marginBottom": "0px",
                "marginLeft": "8px",
                "marginRight": "8px",
                "marginTop": "0px",
                "maxWidth": "70px",
                "minWidth": "50px",
                "textAlign": "center",
              }
            }
            type="number"
            value=""
          />
        </div>
        <p
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#FFFFFF",
              "fontFamily": "Arial",
              "fontSize": "30px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": "34px",
              "marginBottom": "0px",
            }
          }
        >
          )
        </p>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexBasis": "58px",
            "flexDirection": "column",
            "flexGrow": 1,
            "maxWidth": "84px",
            "minWidth": "58px",
            "position": "relative",
          }
        }
      >
        <label
          style={
            Object {
              "color": "#FFFFFF",
              "display": "block",
              "fontFamily": "Arial",
              "fontSize": "12px",
              "fontStyle": "normal",
              "marginBottom": "0px",
              "paddingBottom": "8px",
              "textAlign": "center",
            }
          }
        >
          Exchange
        </label>
        <input
          name="exchange"
          onChange={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          required={true}
          size={3}
          style={
            Object {
              "background": "#FFFAC1",
              "border": "none",
              "borderRadius": "5px",
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "33px",
              "marginBottom": "0px",
              "marginLeft": "8px",
              "marginRight": "8px",
              "marginTop": "0px",
              "maxWidth": "70px",
              "minWidth": "50px",
              "textAlign": "center",
            }
          }
          type="number"
          value=""
        />
      </div>
      <div
        style={
          Object {
            "alignSelf": "flex-end",
            "borderTop": "2px solid #FFFFFF",
            "marginBottom": "18px",
            "marginLeft": "4px",
            "marginRight": "7px",
            "width": "17px",
          }
        }
      />
      <div
        style={
          Object {
            "display": "block",
            "flexBasis": "102px",
            "flexGrow": 4,
            "maxWidth": "138px",
            "minWidth": "102px",
            "position": "relative",
          }
        }
      >
        <label
          style={
            Object {
              "color": "#FFFFFF",
              "display": "block",
              "fontFamily": "Arial",
              "fontSize": "12px",
              "fontStyle": "normal",
              "marginBottom": "0px",
              "paddingBottom": "8px",
              "textAlign": "center",
            }
          }
        >
          Subscriber number
        </label>
        <div
          style={
            Object {
              "display": "flex",
              "width": "100%",
            }
          }
        >
          <input
            name="subscriber1"
            onChange={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            required={true}
            size={2}
            style={
              Object {
                "background": "#FFFAC1",
                "border": "none",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "33px",
                "marginBottom": "0px",
                "marginLeft": "5px",
                "marginRight": "5px",
                "marginTop": "0px",
                "maxWidth": "59px",
                "minWidth": "39px",
                "textAlign": "center",
              }
            }
            type="number"
            value=""
          />
          <input
            name="subscriber2"
            onChange={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            required={true}
            size={2}
            style={
              Object {
                "background": "#FFFAC1",
                "border": "none",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "33px",
                "marginBottom": "0px",
                "marginLeft": "5px",
                "marginRight": "5px",
                "marginTop": "0px",
                "maxWidth": "59px",
                "minWidth": "39px",
                "textAlign": "center",
              }
            }
            type="number"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchBox renders correctly 1`] = `
<div
  style={
    Object {
      "gridArea": "search-box",
      "minWidth": "200px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#315680",
        "backgroundImage": "url(undefined)",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "boxSizing": "border-box",
        "marginLeft": "11px",
        "marginRight": "18px",
        "maxWidth": "633px",
        "minHeight": "260px",
        "paddingLeft": "16px",
        "paddingRight": "16px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "block",
          "paddingTop": "41px",
        }
      }
    >
      <h1
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Georgia",
            "fontSize": "36px",
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginBottom": "22px",
            "textAlign": "center",
          }
        }
      >
        Select a number
      </h1>
      <div
        style={
          Object {
            "borderBottom": "1.3px solid #FFFFFF",
            "opacity": "0.23",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexWrap": "wrap",
          "fontFamily": "Arial",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "justifyContent": "center",
          "margin": "0 auto",
          "paddingTop": "18px",
          "position": "relative",
          "textAlign": "center",
        }
      }
    >
      <p
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#FFFFFF",
            "display": "inline-block",
            "fontSize": "30px",
            "lineHeight": "34px",
            "marginBottom": "0px",
            "marginRight": "10px",
            "verticalAlign": "middle",
          }
        }
      >
        +1
      </p>
      <div
        style={
          Object {
            "display": "flex",
            "flexBasis": "58px",
            "flexGrow": 1,
            "maxWidth": "104px",
            "minWidth": "58px",
            "position": "relative",
          }
        }
      >
        <p
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#FFFFFF",
              "display": "none",
              "fontSize": "12px",
              "left": "-14px",
              "lineHeight": "34px",
              "marginBottom": "0px",
              "position": "absolute",
              "verticalAlign": "middle",
            }
          }
        >
          +1
        </p>
        <p
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#FFFFFF",
              "fontFamily": "Arial",
              "fontSize": "30px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": "34px",
              "marginBottom": "0px",
            }
          }
        >
          (
        </p>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "maxWidth": "84px",
              "width": "100%",
            }
          }
        >
          <label
            style={
              Object {
                "color": "#FFFFFF",
                "display": "block",
                "fontFamily": "Arial",
                "fontSize": "12px",
                "fontStyle": "normal",
                "marginBottom": "0px",
                "paddingBottom": "8px",
                "textAlign": "center",
              }
            }
          >
            Area code
          </label>
          <input
            name="area-code"
            onChange={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            required={true}
            size={3}
            style={
              Object {
                "background": "#FFFAC1",
                "border": "none",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "33px",
                "marginBottom": "0px",
                "marginLeft": "8px",
                "marginRight": "8px",
                "marginTop": "0px",
                "maxWidth": "70px",
                "minWidth": "50px",
                "textAlign": "center",
              }
            }
            type="number"
            value=""
          />
        </div>
        <p
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#FFFFFF",
              "fontFamily": "Arial",
              "fontSize": "30px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": "34px",
              "marginBottom": "0px",
            }
          }
        >
          )
        </p>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexBasis": "58px",
            "flexDirection": "column",
            "flexGrow": 1,
            "maxWidth": "84px",
            "minWidth": "58px",
            "position": "relative",
          }
        }
      >
        <label
          style={
            Object {
              "color": "#FFFFFF",
              "display": "block",
              "fontFamily": "Arial",
              "fontSize": "12px",
              "fontStyle": "normal",
              "marginBottom": "0px",
              "paddingBottom": "8px",
              "textAlign": "center",
            }
          }
        >
          Exchange
        </label>
        <input
          name="exchange"
          onChange={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          required={true}
          size={3}
          style={
            Object {
              "background": "#FFFAC1",
              "border": "none",
              "borderRadius": "5px",
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "33px",
              "marginBottom": "0px",
              "marginLeft": "8px",
              "marginRight": "8px",
              "marginTop": "0px",
              "maxWidth": "70px",
              "minWidth": "50px",
              "textAlign": "center",
            }
          }
          type="number"
          value=""
        />
      </div>
      <div
        style={
          Object {
            "alignSelf": "flex-end",
            "borderTop": "2px solid #FFFFFF",
            "marginBottom": "18px",
            "marginLeft": "4px",
            "marginRight": "7px",
            "width": "17px",
          }
        }
      />
      <div
        style={
          Object {
            "display": "block",
            "flexBasis": "102px",
            "flexGrow": 4,
            "maxWidth": "138px",
            "minWidth": "102px",
            "position": "relative",
          }
        }
      >
        <label
          style={
            Object {
              "color": "#FFFFFF",
              "display": "block",
              "fontFamily": "Arial",
              "fontSize": "12px",
              "fontStyle": "normal",
              "marginBottom": "0px",
              "paddingBottom": "8px",
              "textAlign": "center",
            }
          }
        >
          Subscriber number
        </label>
        <div
          style={
            Object {
              "display": "flex",
              "width": "100%",
            }
          }
        >
          <input
            name="subscriber1"
            onChange={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            required={true}
            size={2}
            style={
              Object {
                "background": "#FFFAC1",
                "border": "none",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "33px",
                "marginBottom": "0px",
                "marginLeft": "5px",
                "marginRight": "5px",
                "marginTop": "0px",
                "maxWidth": "59px",
                "minWidth": "39px",
                "textAlign": "center",
              }
            }
            type="number"
            value=""
          />
          <input
            name="subscriber2"
            onChange={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            required={true}
            size={2}
            style={
              Object {
                "background": "#FFFAC1",
                "border": "none",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "33px",
                "marginBottom": "0px",
                "marginLeft": "5px",
                "marginRight": "5px",
                "marginTop": "0px",
                "maxWidth": "59px",
                "minWidth": "39px",
                "textAlign": "center",
              }
            }
            type="number"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
